lib/socket.dart

```diff
@@ -28,36 +28,36 @@ class SocketIO {
         ) {
     _channel.setMethodCallHandler((call) async {
       if (call.method == PlatformMethod.incomingAck) {
-        var arguments = call.arguments['args'] as List<dynamic>;
-        final reqId = call.arguments['reqId'] as String;
+        var arguments = call.arguments['args'] as List<dynamic>?;
+        final reqId = call.arguments['reqId'] as String?;
         if (reqId == null || !_pendingAcknowledgements.containsKey(reqId)) {
           return;
         }
-        final completer = _pendingAcknowledgements.remove(reqId);
-        arguments = arguments.map(_decodeArgument).toList();
+        final completer = _pendingAcknowledgements.remove(reqId)!;
+        arguments = arguments!.map(_decodeArgument).toList();
         completer.complete(arguments);
       }
     });
   }
 
-  Completer _connectSyncCompleter;
+  Completer? _connectSyncCompleter;
```

Comment:
`late` instead of `?`

Reply:
You set it to null in: -

    void cleanup() {
      onConnectSubscription.cancel();
      onConnectErrorSubscription.cancel();
      _connectSyncCompleter = null;
    }

So it has to have the ?